llvm/lib/Target/AMDGPU/AMDGPULowerKernelArguments.cpp

```diff
@@ -132,6 +132,7 @@ class PreloadKernelArgInfo {
     NF->setAttributes(AL);
     F.replaceAllUsesWith(NF);
     F.setCallingConv(CallingConv::C);
+    F.clearMetadata();
```

Comment:
This should be more targeted. This is going to lose useful information, like `!reqd_work_group_size`

Reply:
I'm a bit lost, isn't any useful metadata already poached by the new function? Is the original function declaration still meaningful in any way?

Reply:
Oh right